src/integration/java/io/pinecone/helpers/GetOrCreateIndex.java

```diff
@@ -0,0 +1,39 @@
+package io.pinecone.helpers;
+
+import io.pinecone.*;
+import io.pinecone.model.CreateIndexRequest;
+import io.pinecone.model.IndexMeta;
+
+import java.io.IOException;
+import java.util.List;
+
+public class GetOrCreateIndex {
```

Comment:
This is the logic for getting an existing index or creating one. This needs to be updated in future when I rigorously test upsert since chances of having an index of dimension d != dimension y for other upserts is high.

Reply:
Imagine a scenario where index an index with a different dimension d exists but my test is expecting dimension y. So I have to now call describeIndexStats to confirm the dimension.

Reply:
Which really makes me wonder if I should create a new index every-time I start a new test so I'm not relying on the previous state.

Reply:
I have updated the logic to check for dimension of the index, otherwise create a new index.